src/main/java/dev/morling/onebrc/CalculateAverage_ebarlas.java

```diff
@@ -21,13 +21,15 @@
 import java.nio.charset.StandardCharsets;
 import java.nio.file.Paths;
 import java.nio.file.StandardOpenOption;
+import java.util.Arrays;
 import java.util.List;
 import java.util.TreeMap;
 
 public class CalculateAverage_ebarlas {
 
-    private static final int HASH_FACTOR = 278;
-    private static final int HASH_MOD = 3_487;
+    private static final int MAX_KEY_SIZE = 64;
+    private static final int HASH_FACTOR = 433;
+    private static final int HASH_TBL_SIZE = 16_383; // range of allowed hash values, inclusive
```

Comment:
As the backing array cannot be grown, this hash table will not be able to handle more than 16k entries.

This again violates the spirit of not making assumptions about the exact values of the data (see this thread https://github.com/gunnarmorling/1brc/discussions/18#discussioncomment-8004520).

And yes, I fell for it too :-)

Reply:
Thanks for pointing that out! However, this Twitter exchange seems to suggest otherwise:

https://twitter.com/royvanrijn/status/1742878555707519363?t=8KIWFDbq6l0mpdgj_BxGhw&s=19

Reply:
![Screenshot_20240104-060031](https://github.com/gunnarmorling/1brc/assets/221014/8d620f02-5759-406b-9e2d-3e8f9ccd6aa0)

Reply:
Yepp, I think it makes sense to limit the maximum number of distinct stations to keep the effort for submissions at check. I've just clarified this in the README, too.

Reply:
Let me know if you have any other concerns about my submission, @gunnarmorling. I believe it is in compliance.

Reply:
Thanks for letting me know about this - now I feel good for not finding time to implement growing and rehashing ;-)

Reply:
> Let me know if you have any other concerns about my submission, @gunnarmorling. I believe it is in compliance.

Awesome, will do. Working on automating the evaluation process right now (it's just too much for doing it manually) and also adding a little test suite. So it will take a little while until I can continue evaluating, but it should hopefully be faster and more thorough and reliable then.

Reply:
Sounds good. I just glanced at the open PRs - there are a lot of them!

I also happened to notice at least one PR which is a copy-paste of my entire prior solution (now out of compliance).

I wonder if it makes sense to add a notice about original work. Otherwise the leaderboard (which is a fun motivator) is sort of meaningless.

Reply:
If it is plain copy/paste it would perform the same, so nothing lost. Perhaps it is also worth declaring ties if times are within 50ms from each other (the range of error).

BTW, looking at your solution, it was strikingly similar to mine (though I used more arrays , you used a data class), so perhaps there are only so many ways to skin that cat ;-)

Reply:
This situation already is discussed in the FAQ:

![Screenshot 2024-01-04 at 16-42-09 gunnarmorling_1brc 1️⃣🐝🏎️ The One Billion Row Challenge -- A fun exploration of how quickly 1B rows from a text file can be aggregated with Java](https://github.com/gunnarmorling/1brc/assets/28612/b839a2bc-9da8-45d2-9d6b-3980662a980d)

I.e. copying in general is ok, but there should be meaningful progress made in addition. If you think there is a submission which goes against that, can you tag me on that PR, ideally with the output of `git diff` between the two files? Thanks!